### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by vogelpi**
This PR affects the documentation only. It contains a major cleanup of the documentation, adds a new block diagram, and slightly adapts a recently added section.

It supersedes PR #66

### CODE REVIEW

File: doc/cs_registers.rst [hunk 1]
```diff
@@ -65,19 +65,20 @@ CSR Address: ``0x300``
...
++-------+-----+---------------------------------------------------------------------------------+
+| 7     | RW  | **Previous Interrupt Enable (MPIE)**, i.e., before entering exception handling. |
++-------+-----+---------------------------------------------------------------------------------+
+| 3     | RW  | **Interrupt Enable (MIE):** If set to 1'b1, interrupts are globally enabled.    |
++-------+-----+---------------------------------------------------------------------------------+
+
+When an exception is encountered, MPIE will be set to MIE.
+When the ``mret`` instruction is executed, the value of MPIE will be stored back to IE.
+
+If you want to enable interrupt handling in your exception handler, set MIE to 1’b1 inside your handler code.
```
Comment (imphil) [inline doc/cs_registers.rst]: nit: looks like you got a wrong apostrophe/single quotation mark in 1'b1?

File: doc/cs_registers.rst [hunk 2]
```diff
@@ -65,19 +65,20 @@ CSR Address: ``0x300``
...
-|       |     | the value of MPIE will be stored to IE.                          |
-+-------+-----+------------------------------------------------------------------+
-| 3     | RW  | **Interrupt Enable:** If you want to enable interrupt handling   |
-|       |     | in your exception handler, set the Interrupt Enable to 1’b1      |
-|       |     | inside your handler code.                                        |
-+-------+-----+------------------------------------------------------------------+
++-------+-----+---------------------------------------------------------------------------------+
+| Bit#  | R/W | Description                                                                     |
++-------+-----+---------------------------------------------------------------------------------+
+| 12:11 | R   | **MPP:** Statically 2’b11 and cannot be altered (read-only).                    |
```
Comment (imphil) [inline doc/cs_registers.rst]: nit: looks like you got a wrong apostrophe/single quotation mark in 2'b11?

File: doc/exception_interrupts.rst [hunk 3]
```diff
@@ -20,26 +26,29 @@ Ibex supports interrupts, exceptions on illegal instructions.
...
+Ibex does support nested interrupt/exception handling.
+Exceptions inside interrupt/exception handlers cause another exception.
+Thus, exceptions during the critical part of your exception handlers, i.e. before having saved the MEPC and MESTATUS registers, will cause those registers to be overwritten.
+Interrupts during interrupt/exception handlers are disabled by default, but can be explicitly enabled if desired.
 
-Upon executing an mret instruction, the core jumps to the program counter saved in the CSR register MEPC and restores the MPIE value of the register MSTATUS to IE. When entering an interrupt/exception handler, the core sets MEPC to the current program counter and saves the current value of MIE in MPIE of the MSTATUS register.
+Upon executing an `mret` instruction, the core jumps to the program counter saved in the CSR register MEPC and restores the MPIE value of the register MSTATUS to IE.
```
Comment (imphil) [inline doc/exception_interrupts.rst]: `` `mret` `` -> `MRET`

File: doc/introduction.rst [hunk 4]
```diff
@@ -6,42 +6,56 @@ Introduction
...
-Much of the code was developed by simplifying the RV32 CPU core "RI5CY" to show how small a RISC-V CPU core could be.
-To make it even smaller, support for the ``E`` exension was added under the code name "MICRO-RISCY".
-Within PULP the core is used as the control core for PULP and PULPino.
+Ibex development started in 2015 under the name "Zero-riscy" as part of the `PULP platform <https://pulp-platform.org>`_ for energy-efficient computing.
+Much of the code was developed by simplifying the RV32 CPU core "RI5CY" to demonstrate how small a RISC-V CPU core could actually be.
+To make it even smaller, support for the "E" exension was added under the code name "Micro-riscy".
+In the PULP ecosystem, the core is used as the control core for PULP, PULPino and PULPissimo.
 
-In December 2018 lowRISC took over the development of ZERO-RISCY and renamed it to Ibex.
+In December 2018 lowRISC took over the development of Zero-riscy and renamed it to Ibex.
```
Comment (imphil) [inline doc/introduction.rst]: Side note: It was previously written ZERO-RISCY (not consistently, but in most places). (No need to change it back, though.)

File: doc/introduction.rst [hunk 5]
```diff
@@ -6,42 +6,56 @@ Introduction
...
 
-Ibex development started in 2015 under the name "ZERO-RISCY" as part of the `PULP platform <http://pulp.ethz.ch/>`_ for
-energy-efficient computing.
-Much of the code was developed by simplifying the RV32 CPU core "RI5CY" to show how small a RISC-V CPU core could be.
-To make it even smaller, support for the ``E`` exension was added under the code name "MICRO-RISCY".
-Within PULP the core is used as the control core for PULP and PULPino.
+Ibex development started in 2015 under the name "Zero-riscy" as part of the `PULP platform <https://pulp-platform.org>`_ for energy-efficient computing.
+Much of the code was developed by simplifying the RV32 CPU core "RI5CY" to demonstrate how small a RISC-V CPU core could actually be.
+To make it even smaller, support for the "E" exension was added under the code name "Micro-riscy".
+In the PULP ecosystem, the core is used as the control core for PULP, PULPino and PULPissimo.
```
Comment (imphil) [inline doc/introduction.rst]: As Davide said, linking the original paper from here would be nice (and much better than in the README). 
https://ieeexplore.ieee.org/document/8106976/ (Is there a free preprint available?)

File: doc/introduction.rst (hunk 4 above)
Comment (imphil) [reply doc/introduction.rst]: Looking at the original paper, it's "Zero-riscy" there. So maybe that's the most official option then :-)

Comment (vogelpi) [reply doc/introduction.rst]: Yep, exaclty. That's why I corrected it in the first place ;-)

File: doc/introduction.rst (hunk 5 above)
Comment (vogelpi) [reply doc/introduction.rst]: Unfortunately, there is no free preprint available. We usually did that. But maybe the conference did not allow it. But I will ask Davide.

Comment (imphil) [reply doc/introduction.rst]: Looks like researchgate has one available. We don't need to link to it, people will find it if they are looking for it.

https://www.researchgate.net/profile/Francesco_Conti4/publication/321121782_Slow_and_steady_wins_the_race_A_comparison_of_ultra-low-power_RISC-V_cores_for_Internet-of-Things_applications/links/5a37aa3ea6fdccdd41fc99e1/Slow-and-steady-wins-the-race-A-comparison-of-ultra-low-power-RISC-V-cores-for-Internet-of-Things-applications.pdf

File: doc/cs_registers.rst (hunk 1 above)
Comment (vogelpi) [reply doc/cs_registers.rst]: Amazing, that you noted that!

**Comment by vogelpi** [comment]
Thanks a lot @imphil , your suggestions are implemented.

**Comment by vogelpi** [comment]
Btw: I also updated the README.md. What is missing

- code of conduct
- reference to our coding styleguide

I think it is crucial that we include a reference to a style guide as early as possible. Thus I included one to the ETH styleguide, which very closely follows ours.

File: README.md [hunk 6]
```diff
@@ -1,26 +1,64 @@
...
+* Do not push to `master`. Instead, create your own branch and then open a Pull Request.
+* Split large contributions into smaller commits addressing individual changes or bug fixes. Do not
+  mix unrelated changes into the same commit!
+* Write meaningful commit messages. For more information, please check out the [contribution
+  guide](https://github.com/lowrisc/ibex/blob/master/CONTRIBUTING.md).
+* If asked to modify your changes, do fixup your commits and rebase your branch to maintain a
+  clean history.
+
+When contributing SystemVerilog source code, please try to be consistent, e.g., by adhering to the
+following [coding style guidelines](https://github.com/pulp-platform/style-guidelines).
```
Comment (imphil) [inline README.md]: This style guide doesn't fully match ours, so let's just remove it for now and go with the standard advice for all projects: Please match the style of the existing code.

Comment (vogelpi) [reply README.md]: I changed the sentence to match the existing code if possible. But I kept a reference to the other style guide. It is definitely better than nothing and still short.

**Comment by sjgitty** [comment]
I'm trying to get approval for our lowRISC style guide to go "live" today,
discussions underway

**Pull request merged**